content/en/docs/instrumentation/js/serverless.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Serverless
+weight: 8
+---
+
+This guide will show you how to get started with tracing serverless functions using OpenTelemetry instrumentation libraries.
+
+## AWS Lambda
+
+Following the next steps will allow your AWS Lambda functions to use OpenTelemetry to send traces to the configured backend.
+
+### Dependencies
+
+Create an empty package.json:
+
+```sh
+npm init -y
+```
+
+Install dependencies used by the example.
+
+```sh
+npm install \
+  @opentelemetry/api \
+  @opentelemetry/auto-instrumentations-node \
+  @opentelemetry/exporter-trace-otlp-http \
+  @opentelemetry/instrumentation \
+  @opentelemetry/sdk-trace-base \
+  @opentelemetry/sdk-trace-node
+```
+
+### AWS Lambda wrapper code
+
+This file will contain all the OpenTelemetry logic, which will enable tracing.
+Please save the following code as `lambda-wrapper.js`.
+
+```javascript
+/* lambda-wrapper.js */
+
+const api = require("@opentelemetry/api");
+const { BatchSpanProcessor } = require("@opentelemetry/sdk-trace-base");
+const {
+  OTLPTraceExporter,
+} = require("@opentelemetry/exporter-trace-otlp-http");
+const { NodeTracerProvider } = require("@opentelemetry/sdk-trace-node");
+const { registerInstrumentations } = require("@opentelemetry/instrumentation");
+const {
+  getNodeAutoInstrumentations,
+} = require("@opentelemetry/auto-instrumentations-node");
+
+api.diag.setLogger(new api.DiagConsoleLogger(), api.DiagLogLevel.ALL);
+
+const provider = new NodeTracerProvider();
+const collectorOptions = {
+  url: "<backend_url>",
+};
+
+const spanProcessor = new BatchSpanProcessor(
+  new OTLPTraceExporter(collectorOptions)
+);
+
+provider.addSpanProcessor(spanProcessor);
+provider.register();
+
+registerInstrumentations({
+  instrumentations: [
+    getNodeAutoInstrumentations({
+      "@opentelemetry/instrumentation-aws-lambda": {
+        disableAwsContextPropagation: true,
+      },
+    }),
+  ],
+});
+```
+
+Replace `<backend_url>` with the URL of your favorite backend to export all
+traces to it. If you don't have one setup already, you can check out [Jaeger](https://www.jaegertracing.io/) or [Zipkin](https://zipkin.io/).
+
+Note that `disableAwsContextPropagation` is set to true. The reason for this is
+that the Lambda instrumentation tries to use the X-Ray context headers by
+default, this results in a non-sampled context, which creates a
+`NonRecordingSpan`.
+
+More details can be found in the instrumentation
+[documentation](https://www.npmjs.com/package/@opentelemetry/instrumentation-aws-lambda).
+
+### AWS Lambda function handler
+
+Now that you have a Lambda wrapper, you can create a simple handler that will
+serve as a Lambda function. Save the following code as `handler.js`.
+
+```javascript
+/* handler.js */
+
+"use strict";
+
+const https = require('https');
+
+function getRequest() {
```

Comment:
Is `async` and `await` possible to use in lambda? Would prefer to show that instead of promises if possible

Reply:
I went with this as a reference since I want to wait for the response and return it
https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html#nodejs-handler-async